### PR TITLE
[WFLY-6971] : MDB cannot be deployed after migration due to invalid cache configuration. Log a message in order to detect and reject any invalid configuration.

### PR DESCRIPTION
**Pull request by panossot**
[WFLY-6971] : https://issues.jboss.org/browse/WFLY-6971

### CODE REVIEW

**Comment by pferraro** [comment]
This is the wrong approach.  See my comments in jira.

File: ejb3/src/main/java/org/jboss/as/ejb3/cache/distributable/DistributableCacheFactoryBuilderService.java [hunk 1]
```diff
@@ -95,6 +98,18 @@ public void installDeploymentUnitDependencies(ServiceTarget target, ServiceName
     }
 
     @Override
+    public void installDeploymentUnitDependencies(ServiceTarget target, ServiceName deploymentUnitServiceName, List<String> builderServices) {
+        for (Builder<?> builder : this.builder.getDeploymentBuilders(deploymentUnitServiceName)) {
+            if (!((ArrayList<String>)builderServices).contains(builder.getServiceName().getCanonicalName())) {
```
Comment (panossot) [inline ejb3/src/main/java/org/jboss/as/ejb3/cache/distributable/DistributableCacheFactoryBuilderService.java]: @pferraro The collection of the names of the services in used here, in order to detect the duplicate ones.

**Comment by kabir** [comment]
@pferraro please look now

**Comment by kabir** [comment]
@pferraro @panossot Should this be closed?

**Comment by panossot** [comment]
I will at least try to remove the conflicts.

File: ejb3/src/main/java/org/jboss/as/ejb3/cache/distributable/DistributableCacheFactoryBuilderService.java (hunk 1 above)
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/cache/distributable/DistributableCacheFactoryBuilderService.java]: Why are you casting to an ArrayList?

**Comment by pferraro** [comment]
The general approach here is still misguided.  If there are duplicate services being installed - then there is a problem with the configuration.  Merely skipping the installation of services that already exist will only mask a deeper problem.
A proper fix should encompass the add handler for the passivation store that rejects an attempt to add a passivation store containing configuration that duplicates another passivation store.

**Comment by panossot** [comment]
@pferraro : Could you review again, please?

**Comment by panossot** [comment]
Closing this PR.

**Pull request closed without merging**